recipes/ptychodus/meta.yaml

```
@@ -0,0 +1,60 @@
+{% set name = "ptychodus" %}
+{% set version = "0.2.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/AdvancedPhotonSource/ptychodus/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: f43b86741f341176133b350474e8511c66e4e26938d3b2592d3c5ec324f4e6f9
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - ptychodus = ptychodus.__main__:main
+
+requirements:
+  host:
+    - pip
+    - python >=3.9
+    - setuptools >=46.4.0
+    - setuptools_scm[toml] >=6.2
+    - setuptools_scm_git_archive
+    - wheel
+  run:
+    - h5py =3.*
+    - hdf5-external-filter-plugins-lz4
+    - matplotlib-base =3.*
+    - numpy =1.*
+    - python >=3.9
+    - scipy =1.*
+    - watchdog =2.*
+  run_constrained:
+    - pyqt =5.*
+    - tike =0.22.*
```

```suggestion
  run:
    - h5py =3.*
    - hdf5-external-filter-plugins-lz4
    - matplotlib-base =3.*
    - numpy =1.*
    - pyqt =5.*
    - python >=3.9
    - scipy =1.*
    - watchdog =2.*
  run_constrained:
    - tike =0.22.*
```
Moving `pyqt` from optional to required sections because even though opening a GUI is not required, importing a bunch of QT classes and functions is required to run `ptychodus`.

---

https://github.com/AdvancedPhotonSource/ptychodus/pull/34